src/Logary/Registry.fs

```diff
@@ -207,6 +214,60 @@ module Registry =
          && not ("true" = Env.varDefault "LOGARY_I_PROMISE_I_HAVE_PURCHASED_LICENSE" (fun () -> "false")) then
          failwith "You must purchase a license for Logary to run it on or with IIS or Kestrel."
 
+    // check if msg has enable metric counter
+    let hookupMetric (metricRegistry: MetricRegistry) (message: Message) =
+      // process log message as counter
+      let counterMetricConf = message |> Message.tryGetCounterMetricConf
+      match counterMetricConf with
+      | None -> do ()
+      | Some counterMetricConf ->
+        let counterMetric = GaugeConf.create counterMetricConf |> metricRegistry.registerMetric
+        let counter =
+          match counterMetricConf.labelNames with
+          | [||] -> counterMetric.noLabels
+          | labelNames ->
+            // find lable name's value from message's context, fields goes first
+            let labelValues =
+              labelNames |> Array.map (fun labelName ->
+                match message |> Message.tryGetField labelName with
+                | Some labelValue -> string labelValue
+                | None ->
+                  match message |> Message.tryGetContext labelName with
+                  | Some labelValue -> string labelValue
+                  | None -> failwithf "label name (%s) not found in message,maybe you should metric with nolables" labelName
+              )
+            counterMetric.labels labelValues
+
+        counter.inc 1.
+
+
+      // process logary's gauge
+      let sensorName = message.name |> PointName.format
+      let gauges = message |> Message.getAllGauges
+      gauges |> Seq.iter (fun (gaugeName, gauge) ->
```

Comment:
Can't we just have the metric instance match towards the point name instead of the above "counter metric"?

Reply:
here is for hook up logary's `Gauge` with prometheus's gauge, legacy code using  `gaugeWithUnit` in `MessageModule` api, can get this benefit by one configuration as logary's initiation.

Reply:
since message know nothing about the registry/logger, but metric needs register on some `MetricRegistry` which in a logary's `LogManager/Registry`, so here we need store info in message, when dispatch by registry, grab from it and hook with `metricRegistry`